### ISSUE TITLE
Editing front page without a posts page set: "Sorry, you are not allowed to edit this item"

### ISSUE THREAD

**Issue by supernovia** [Automattic/wp-calypso#53167]
### Steps to reproduce the behavior
I have not been able to duplicate this on a test site yet, but we have a handful of reports now. Here are the symptoms:

- Click edit on the front (static) page. Note this is happening to users AND superadmins, so editor does not seem to matter.
- See the error "Sorry, you are not allowed to edit this item."
- If you view /wp-admin/edit.php?post_type=page , notice the page has an ℹ️  icon and says it is the posts page
- Go to Customize. Note this page is NOT set as the posts page, and in fact no posts page is specified.

#### What I expected to happen
If no posts page is set, I shouldn't have trouble editing any site pages.

#### What actually happened
The site seems to think the front page is the posts page. It is not.

If we set up a new posts page and specify that in Customize, the problem goes away.

### Context
This came up in a few different tickets and forums threads.  Here is the one where I realized it's an issue:
3960837-zd-woothemes

I'll add others.

#### Browser / OS version
Doesn't matter.

#### Is this specific to the applied theme? Which one?
Does not seem to be.

#### Does this happen on simple or atomic sites or both?
So far I have seen it only on Simple Sites

#### Is there any console output or error text?
GET https://sitename.wordpress.com/wp-admin/post.php?post=1&action=edit 500

The ID and site name will be different, obviously.

#### Level of impact
So far I'm aware of it impacting four sites.

#### Reproducibility
I do not not yet know how to reproduce this.

#### Screenshot / Video: If applicable, add screenshots to help explain your problem.

Customizer has a front page set:
![Screen Shot 2021-05-25 at 5 05 13 PM](https://user-images.githubusercontent.com/3476361/119579469-a615b680-bd7b-11eb-9755-548dbba60cef.png)

We get this error if we click edit:
![Screen Shot 2021-05-25 at 5 05 35 PM](https://user-images.githubusercontent.com/3476361/119579535-d198a100-bd7b-11eb-9a3f-8b88b1aab309.png)

The wp-admin Ppages list says this page is set as the posts page, even though it is not:
![Uploading Screen Shot 2021-05-25 at 5.06.00 ![Screen Shot 2021-05-25 at 5 06 05 PM](https://user-images.githubusercontent.com/3476361/119579604-fee54f00-bd7b-11eb-8183-99f3fdea9f84.png)M.png…]()

**Comment by supernovia** [comment]
Another case here:
https://wordpress.com/forums/topic/unable-to-edit-page-2/

**Comment by kylemcph** [comment]
User reached out in 29787241-hc with the same issue. I was able to replicate on a test WPcom simple site. 

Steps to replicate:
1. Customize > Homepage Settings > set a page as posts page, save changes. 
2. Customize > Homepage Settings > change setting to display latest posts
3. Try to update the page set as posts page in step 1. This produces the same Sorry, you are not allowed to edit this item. error message.
4. A temporary fix is to adjust homepage settings, remove page you want to edit as post page, and set homepage settings back to display latest posts.

This is not theme-related as I tried with multiple themes and always reproduced the same error message.
